### PR TITLE
Add `Iterator.prototype.toAsync`

### PR DESCRIPTION
**Pull request by zloirock**
Resolves https://github.com/tc39/proposal-iterator-helpers/issues/160#issuecomment-984635413

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -602,6 +602,17 @@ <h1>Iterator.prototype.toArray ( )</h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-iteratorprototype.toasync">
+        <h1>Iterator.prototype.toAsync ( )</h1>
+        <emu-alg>
+          1. Let _syncIteratorRecord_ be ? GetIteratorDirect(*this* value).
```
Comment (devsnek) [inline spec.html:608]: maybe hot take
`1. Return ? Call(%AsyncIterator.from%, %AsyncIterator%, &laquo; *this* value &raquo;).`

**Comment by zloirock** [comment]
@bakkot @michaelficarra hey, maybe makes sense to review it before this TC39 meeting? @codehag wrote about adding it in the next round.

**Comment by bakkot** [comment]
@zloirock We're going to get feedback from the committee about including this method; if the committee in general is in favor we'll review/land it after that.

File: spec.html [hunk 2]
```diff
@@ -603,6 +603,17 @@ <h1>Iterator.prototype.toArray ( )</h1>
...
+          1. Let _asyncIteratorRecord_ to CreateAsyncFromSyncIterator(_syncIteratorRecord_).
```
Comment (michaelficarra) [inline spec.html:610]:
```suggestion
          1. Let _asyncIteratorRecord_ be CreateAsyncFromSyncIterator(_syncIteratorRecord_).
```

**Review by michaelficarra — APPROVED** [review]
The committee was overall positive about this, so we will include it. Spec text LGTM.

**Pull request merged**